docs/training-apis/api-core/hpsearch.txt

```diff
@@ -0,0 +1,79 @@
+.. _core-hpsearch:
+
+#######################
+ Hyperparameter Search
+#######################
+
+With Core API you can run advanced hyperparameter searches with arbitrary training code. The
+hyperparameter search logic is in the master, which coordinates many different Trials. Each trial
+runs a train-validate-report loop:
+
+.. table::
+
+   +----------+--------------------------------------------------------------------------+
+   | Train    | Train until a point chosen by the hyperparameter search algorithm and    |
+   |          | obtained via the Core API.  The length of training is absolute, so you   |
+   |          | have to keep track of how much you have already trained in order to now  |
+   |          | how much more to train.                                                  |
+   +----------+--------------------------------------------------------------------------+
+   | Validate | Validate your model to obtain the metric you configure in the            |
+   |          | ``searcher.metric`` field of your experiment config.                     |
+   +----------+--------------------------------------------------------------------------+
+   | Report   | Use the Core API to report results to the master.                        |
+   +----------+--------------------------------------------------------------------------+
+
+Create a ``3_hpsearch.py`` training script by copying the ``2_checkpoints.py`` script you created in
+:ref:`core-checkpoints`.
+
+In your ``if __name__ == "__main__"`` block, access the hyperparameter values chosen for this trial
+using the ClusterInfo API and configure the training loop accordingly:
+
+.. literalinclude:: ../../../examples/tutorials/core_api/3_hpsearch.py
+   :language: python
+   :dedent:
+   :start-at: hparams = info.trial.hparams
+
+Next modify ``main()`` to run the train-validate-report loop mentioned above by iterating through
```

Comment:
Add a comma after "Next"

Reply:
removing Next since you requested numbering the steps